java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -1369,6 +1380,9 @@ open class KotlinUsesExtractor(
 
     fun getTypeParameterParentLabel(param: IrTypeParameter) =
         param.parent.let {
+            (it as? IrDeclaration)?.let { decl ->
+                overriddenFunctionAttributes[decl]?.id
+            } ?:
```

Comment:
Doesn't this and all other `overriddenFunctionAttributes` uses in `KotlinUsesExtractor` assume that when the access is made, we're extracting the given function with overrides? So in this case, are we sure that `getTypeParameterParentLabel` is only called when `overriddenFunctionAttributes[decl]` is being extracted? 

I think, you might be right that `getTypeParameterParentLabel` and `getValueParameterLabel` are only called within `forceExtractFunction` and `extractGeneratedOverload`, but it feels somewhat fragile. Isn't this the same as `(this as KotlinFileExtractor).declarationStack.peek().overriddenFunctionAttributes`, which doesn't make sense at the moment, because the attributes are not on the stack, but if they were, we would consider this dangerous because of the cast.

Reply:
For value parameters: I'd expect this could be seen from inside capturing local declarations and classes too. Something like

```
fun f(x: Int, y: Int = x.let {
  fun captures() = it
  captures()
})
```

For type parameters: these definitely can be seen from elsewhere, principally when extracting instantiated generic types that occur in the signature of a method with overloads. So when we extract `fun <T> f(x: List<T>, y: T? = null)` then we need to generate `List<T>` with `T = f/1::T` as well as the case with `T = f/2::T` that gets created for the "normal" version of `f`. That's why the overridden functions map gets passed to child instances of KotlinUses/FileExtractor.